benchmarks/inference-server/triton/README.md

```diff
@@ -0,0 +1,198 @@
+# AI on GKE: Benchmark TensorRT LLM on Triton Server
+
+This guide provides instructions for deploying and benchmarking a TensorRT Large Language Model (LLM) on Triton Inference Server within a Google Kubernetes Engine (GKE) environment. The process involves building a Docker container with the TensorRT LLM engine and deploying it to a GKE cluster. 
+
+## Prerequisites
+
+- Docker
+- Google Cloud SDK
+- Kubernetes CLI (kubectl)
+- Hugging Face account for model access
+- NVIDIA GPU drivers and CUDA toolkit (to build the TensorRTLLM Engine)
+
+## Step 1: Build the TensorRT LLM Engine and Docker Image
+
+1. **Build the TensorRT LLM Engine:** Follow the instructions provided in the [TensorRT LLM backend repository](https://github.com/triton-inference-server/tensorrtllm_backend/blob/main/README.md) to build the TensorRT LLM Engine.
+
+2. **Setup the Docker container:**
+   
+   ***Method 1: Add the Model repository and the relevant scripts to the image***
+
+   Inside the `tritonllm_backend` directory, create a Dockerfile with the following content ensuring the Triton Server is ready with the necessary models and scripts.
+
+   ```Dockerfile
+   FROM nvcr.io/nvidia/tritonserver:23.10-trtllm-python-py3
+
+   # Copy the necessary files into the container
+   COPY all_models /all_models
+   COPY scripts /opt/scripts
+
+   # Install Python dependencies
+   RUN pip install sentencepiece protobuf
+
+   RUN chmod +x /opt/scripts/start_triton_servers.sh
+
+   CMD ["/opt/scripts/start_triton_servers.sh"]
+   ```
+
+   The Shell script `/opt/scripts/start_triton_servers.sh` is like below: 
+
+   ```start_triton_servers.sh
+   #!/bin/bash
+
+
+    # Use the environment variable to log in to Hugging Face CLI
+    huggingface-cli login --token $HUGGINGFACE_TOKEN
+
+
+
+
+    # Launch the servers (modify this depending on the number of GPU used and the exact path to the model repo)
+    mpirun --allow-run-as-root -n 1 /opt/tritonserver/bin/tritonserver \
```

Comment:
in this particular example I am running two instances of triton server with different shared memory region prefix name as the engine was built for 2 GPUs. Will this still work without mpirun?

Reply:
Yes, "-n 1" essentially means we are running 1 instance. Multi-GPU should still work since a single Triton process can use multiple GPUs with Tensor parallelism.